client-js/rtvi/messages.ts

```diff
@@ -0,0 +1,226 @@
+  version as packageVersion,
+} from "../package.json";
+
+export const RTVI_PROTOCOL_VERSION = "1.0.0";
+export const RTVI_MESSAGE_LABEL = "rtvi-ai";
+
+/**
+ * Messages the corresponding server-side client expects to receive about
+ * our client-side state.
+ */
+export enum RTVIMessageType {
+  /** Outbound Messages */
+  CLIENT_READY = "client-ready",
+  DISCONNECT_BOT = "disconnect-bot",
+  // Client-to-server messages
+  CLIENT_MESSAGE = "client-message",
+  APPEND_TO_CONTEXT = "append-to-context",
+
+  /**
+   * Inbound Messages
+   * Messages the server-side client sends to our client-side client regarding
+   * its state or other non-service-specific messaging.
+   */
+  BOT_READY = "bot-ready", // Bot is connected and ready to receive messages
+  ERROR = "error", // Bot initialization error
+  METRICS = "metrics", // Bot reporting metrics
+  SERVER_MESSAGE = "server-message", // Custom server-to-client message
+  SERVER_RESPONSE = "server-response", // Server response to client message
+  ERROR_RESPONSE = "error-response", // Error message in response to an outbound message
+  APPEND_TO_CONTEXT_RESULT = "append-to-context-result", // Result of appending to context
+
+  /** Transcription Messages */
+  USER_TRANSCRIPTION = "user-transcription", // Local user speech to text transcription (partials and finals)
+  BOT_TRANSCRIPTION = "bot-transcription", // Bot full text transcription (sentence aggregated)
+  USER_STARTED_SPEAKING = "user-started-speaking", // User started speaking
+  USER_STOPPED_SPEAKING = "user-stopped-speaking", // User stopped speaking
+  BOT_STARTED_SPEAKING = "bot-started-speaking", // Bot started speaking
+  BOT_STOPPED_SPEAKING = "bot-stopped-speaking", // Bot stopped speaking
+
+  /** LLM Messages */
+  USER_LLM_TEXT = "user-llm-text", // Aggregated user input text which is sent to LLM
+  BOT_LLM_TEXT = "bot-llm-text", // Streamed token returned by the LLM
+  BOT_LLM_STARTED = "bot-llm-started", // Bot LLM inference starts
+  BOT_LLM_STOPPED = "bot-llm-stopped", // Bot LLM inference stops
+
+  // Function calling
+  LLM_FUNCTION_CALL = "llm-function-call", // Inbound function call from LLM
+  LLM_FUNCTION_CALL_RESULT = "llm-function-call-result", // Outbound result of function call
+
+  BOT_LLM_SEARCH_RESPONSE = "bot-llm-search-response", // Bot LLM search response
+
+  /** TTS Messages */
+  BOT_TTS_TEXT = "bot-tts-text", // Bot TTS text output (streamed word as it is spoken)
+  BOT_TTS_STARTED = "bot-tts-started", // Bot TTS response starts
+  BOT_TTS_STOPPED = "bot-tts-stopped", // Bot TTS response stops
+}
+
+// ----- Message Data Types
+
+export type BotReadyData = {
+  version: string;
+  about?: unknown; // Optional about data from the bot
+};
+
+type PlatformDetailsValue = undefined | string | number | boolean;
+type NestedPlatformDetails =
+  | PlatformDetailsValue
+  | Record<string, PlatformDetailsValue>;
+
+// This is an interface so that different client libraries can provide their own
```

Comment:
by "different client libraries" you mean only this one (the web one) and the react native one, which is based on this one?

Reply:
js, react, and react native